### PR TITLE
imp: add flat fee for queries to `VerifyMembership`

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

see https://github.com/cosmos/cosmos-sdk/pull/9995 for similar flat gas consumption fee logic.

closes: #5902 

i elected to consume the gas outside of the defer func as we are not operating on the cached context in this case, and simply consuming gas for the read on consensus state from the store...but open to other opinions if people have any!

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Refactor**
	- Improved gas consumption logic for proof verification queries to ensure proper gas handling.

### CODE REVIEW

File: modules/core/02-client/keeper/grpc_query.go [hunk 1]
```diff
@@ -386,6 +387,12 @@ func (k Keeper) VerifyMembership(c context.Context, req *types.QueryVerifyMember
 		return nil, status.Error(codes.FailedPrecondition, errorsmod.Wrapf(types.ErrClientNotActive, "cannot verify membership using client (%s) with status %s", req.ClientId, clientStatus).Error())
 	}
 
+	// consume flat gas fee for queries to this rpc to read consensus state bytes from store
```
Comment (damiannolan) [inline modules/core/02-client/keeper/grpc_query.go:390]:
```suggestion
	// consume flat gas fee for proof verification queries
```

**Review by crodriguezvega — APPROVED** [review]
🙏

File: modules/core/02-client/keeper/grpc_query.go [hunk 2]
```diff
@@ -386,6 +387,12 @@ func (k Keeper) VerifyMembership(c context.Context, req *types.QueryVerifyMember
 		return nil, status.Error(codes.FailedPrecondition, errorsmod.Wrapf(types.ErrClientNotActive, "cannot verify membership using client (%s) with status %s", req.ClientId, clientStatus).Error())
 	}
 
+	// consume flat gas fee for proof verification queries
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/02-client/keeper/grpc_query.go:390]: Personally I think its fine to just state that we're consuming a flat fee for proof verification, as its cpu intensive and  not based around IOPS, which is the main gas consumption factor in the cosmos sdk. 

Pretty sure all `VerifyMembership` calls within the client state will have to look up a root to prove against based on the provided height, so there would be gas consumed on the `cacheCtx` for that store access which should also protection against recursive client calls, but this just adds an extra bit of gas consumption which is definitely no harm and the right call imo!

[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/02-client/keeper/grpc_query.go:390]: You're 100% correct! Great point! 

Happy to add a note if others agree too, maybe something like:

```suggestion
	// consume flat gas fee for proof verification queries.
	// NOTE: consuming gas prior to method invocation also provides protection against recursive calls reaching stack overflow
```

**Review by damiannolan — APPROVED** [review]

File: modules/core/02-client/keeper/grpc_query.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/core/02-client/keeper/grpc_query.go]: great suggestion! ill update this, thanks for the reviews and reasonings @damiannolan @DimitrisJim

**Pull request merged**